apps/api/src/app/contexts/contexts.controller.ts

```diff
@@ -0,0 +1,184 @@
+} from './dtos';
+import { CreateContext, CreateContextCommand } from './usecases/create-context';
+import { DeleteContext, DeleteContextCommand } from './usecases/delete-context';
+import { GetContext, GetContextCommand } from './usecases/get-context';
+import { GetContexts, GetContextsCommand } from './usecases/get-contexts';
+import { UpdateContext, UpdateContextCommand } from './usecases/update-context';
+
+@Controller('/contexts')
+@UseInterceptors(ClassSerializerInterceptor)
+@ThrottlerCategory(ApiRateLimitCategoryEnum.GLOBAL)
+@RequireAuthentication()
+@ApiTags('Contexts')
+@ApiCommonResponses()
+@ApiExcludeController()
+export class ContextsController {
+  constructor(
+    private createContextUsecase: CreateContext,
+    private getContextUsecase: GetContext,
+    private getContextsUsecase: GetContexts,
+    private updateContextUsecase: UpdateContext,
+    private deleteContextUsecase: DeleteContext,
+    private featureFlagsService: FeatureFlagsService
+  ) {}
+
+  private async checkFeatureEnabled(user: UserSessionData) {
+    const isEnabled = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.IS_CONTEXT_ENABLED,
+      defaultValue: false,
+      organization: { _id: user.organizationId },
+      user: { _id: user._id },
+    });
+
+    if (!isEnabled) {
+      throw new ForbiddenException('Context feature is not enabled');
+    }
+  }
+
+  @Post('')
+  @ApiResponse(GetContextResponseDto, 201)
+  @ApiOperation({
+    summary: 'Create context',
+    description: 'Create a new context with the specified type, key, and data',
+  })
+  @ExternalApiAccessible()
+  async createContext(
+    @UserSession() user: UserSessionData,
+    @Body() body: CreateContextRequestDto
+  ): Promise<GetContextResponseDto> {
+    await this.checkFeatureEnabled(user);
+
+    return this.createContextUsecase.execute(
+      CreateContextCommand.create({
+        organizationId: user.organizationId,
+        environmentId: user.environmentId,
+        type: body.type,
+        identifier: body.identifier,
+        data: body.data,
+      })
+    );
+  }
+
+  @Get('')
+  @ApiResponse(GetContextsResponseDto)
+  @ApiOperation({
+    summary: 'Get contexts',
+    description: 'Retrieve a paginated list of contexts, optionally filtered by type and key pattern',
+  })
+  @ExternalApiAccessible()
+  async getContexts(
+    @UserSession() user: UserSessionData,
+    @Query() query: GetContextsRequestDto
+  ): Promise<GetContextsResponseDto> {
+    await this.checkFeatureEnabled(user);
+
+    return this.getContextsUsecase.execute(
+      GetContextsCommand.create({
+        user,
+        limit: query.limit || 10,
+        after: query.after,
+        before: query.before,
+        orderDirection: query.orderDirection,
+        orderBy: query.orderBy || 'createdAt',
+        includeCursor: query.includeCursor,
+        type: query.type,
+        identifier: query.identifier,
+      })
+    );
+  }
+
+  @Get('/:identifier')
```

Comment:
I think we discussed doing a double param here? 


```suggestion
  @Get('/:type/:identifier')
```